core/tests/protractor_utils/users.js

```diff
@@ -61,24 +61,34 @@ var login = async function(email, useManualNavigation = true) {
     await browser.get(general.SERVER_URL_PREFIX + general.LOGIN_URL_SUFFIX);
   }
 
+
   var loginPage = element(by.css('.protractor-test-login-page'));
   await waitFor.presenceOf(loginPage, 'Login page did not load');
 
   var emailInput = element(by.css('.protractor-test-sign-in-email-input'));
   await action.sendKeys('Email input', emailInput, email);
 
   var signInButton = element(by.css('.protractor-test-sign-in-button'));
-  await action.click('Sign in button', signInButton);
 
-  await waitFor.pageToFullyLoad();
+  await waitFor.clientSideRedirection(async() => {
+    // Clicking sign in button to trigger redirection.
+    await action.click('Sign in button', signInButton);
+  }, (URL) => {
+    // Waiting till url is not /login anymore (that is url is changed).
+    return !(/login/.test(URL));
+  });
 };
 
 var logout = async function() {
   await browser.get(general.SERVER_URL_PREFIX + general.LOGOUT_URL_SUFFIX);
-  // Wait for logout page to load.
-  await waitFor.pageToFullyLoad();
-  // Wait for redirection to occur.
-  await waitFor.pageToFullyLoad();
+
+  await waitFor.clientSideRedirection(async() => {
+    // Once logout page is reloaded, it triggers redirection.
+    await waitFor.pageToFullyLoad();
+  }, (URL) => {
+    // Waiting till url is not /logout anymore (that is url is changed)
```

Comment:
Wait until the URL has changed to something that is not /logout.

Reply:
done